Incompatibility problem with Gradle 9.0-rc1 not finding class DefaultDecoratedConvention

Hi,

This plugin is currently not compatible with upcoming Gradle 9.0 version due to `ClassNotFoundException: org.gradle.api.plugins.internal.DefaultDecoratedConvention`.

I am not familiar with this, but I believe this is due to a major change in Gradle removing built-in core conventions ([see this issue](https://github.com/gradle/gradle/issues/22056) that has been delivered on Gradle 9.0) which were marked as deprecated back in [Gradle 8.2](https://docs.gradle.org/8.2/release-notes.html).


Sharing the stacktrace.
```
* Exception is:
org.gradle.api.ProjectConfigurationException: A problem occurred configuring project ':demo-project:app'.
        at org.gradle.configuration.project.LifecycleProjectEvaluator.wrapException(LifecycleProjectEvaluator.java:84)
...
...
...
Caused by: java.lang.ClassNotFoundException: org.gradle.api.plugins.internal.DefaultDecoratedConvention
        at org.gradle.internal.classpath.transforms.ConventionInterceptors.interceptProjectGetConventionMethod(ConventionInterceptors.java:44)
        at io.github.gmazzo.android.test.aggregation.InternalDSLKt.getAndroid(InternalDSL.kt:116)
        at io.github.gmazzo.android.test.aggregation.AndroidTestCoverageAggregationPlugin$apply$1$1.invoke$lambda$0(AndroidTestCoverageAggregationPlugin.kt:51)
```

--- a/plugin/src/main/kotlin/io/github/gmazzo/android/test/aggregation/InternalDSL.kt
+++ b/plugin/src/main/kotlin/io/github/gmazzo/android/test/aggregation/InternalDSL.kt
@@ -19,15 +19,15 @@ import org.gradle.kotlin.dsl.create
 import org.gradle.kotlin.dsl.findByType
 import org.gradle.kotlin.dsl.get
 import org.gradle.kotlin.dsl.getByName
+import org.gradle.kotlin.dsl.getByType
 import org.gradle.kotlin.dsl.named
 import org.gradle.kotlin.dsl.testAggregation
-import org.gradle.kotlin.dsl.the
 import org.gradle.testing.jacoco.plugins.JacocoTaskExtension
 import org.gradle.util.GradleVersion
 import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
 
 internal val Project.android
-    get() = the<BaseExtension>()
+    get() = extensions.getByName<BaseExtension>("android")
 
 internal val Project.androidComponents
     get() = extensions.getByName<AndroidComponentsExtension<*, *, *>>("androidComponents")
@@ -107,6 +107,6 @@ internal val TaskProvider<AbstractTestTask>.execData
     get() = map {
         when (it) {
             is AndroidUnitTest -> it.jacocoCoverageOutputFile
-            else -> it.the<JacocoTaskExtension>().destinationFile
+            else -> it.extensions.getByType<JacocoTaskExtension>().destinationFile
         }
     }

